pygdf/buffer.py

```diff
@@ -63,10 +63,12 @@ def __sizeof__(self):
     def __getitem__(self, arg):
         if isinstance(arg, slice):
             sliced = self.to_gpu_array()[arg]
-            return Buffer(sliced)
+            buf = Buffer(sliced)
+            buf.dtype = self.dtype  # for np.datetime64 support
+            return buf
         elif isinstance(arg, int):
             arg = utils.normalize_index(arg, self.size)
-            return self.mem[arg]
+            return self.mem[arg].view(dtype=self.dtype)
```

Comment:
I'm running this branch with the latest libgdf master. This change is causing me to get `TypeError: view() takes no keyword arguments`, not sure why. Could there any dependency problem, mismatch in the branches or something else?

Reply:
`self.mem[arg]` ought to be numpy scalar instance. What numpy and pandas versions are you using?

Reply:
numpy 1.12.1 and pandas 0.23.3

Reply:
can you show us how to repro?

Reply:
Even the simples example that I got from `test_datetime` gives me the error: 

```
import pandas as pd
import numpy as np
from pygdf import DataFrame, Series

idx = pd.date_range('20010101', '20020215', freq='400h')

pd_data = pd.Series(idx.copy())
gdf_data = Series(pd_data)
gdf_data #here I get it.
```
Also, quite a few of the tests fail, if they pass for others (with libgdf-master) I'm guessing the problem is somewhere in my conda environment.

Reply:
i don't think it's just a numpy version issues since the dtype parameter was in numpy 1.12

https://docs.scipy.org/doc/numpy-1.12.0/reference/generated/numpy.ndarray.view.html

So something else must be going on

thanks

Reply:
something weird with numpy 1.12

ipdb> self.mem[arg].view(dtype=self.dtype)
*** TypeError: view() takes no keyword arguments
ipdb> self.mem[arg].view(self.dtype)
numpy.datetime64('2001-01-01T00:00:00.000')
ipdb> type(self.mem[arg])
<class 'numpy.int64'>
ipdb> np.int64(2).view(dtype=self.dtype)
*** TypeError: view() takes no keyword arguments
ipdb> np.int64(2).view(self.dtype)
numpy.datetime64('1970-01-01T00:00:00.002')

@pearu it looks like it works if we don't pass it as a kwarg... which is strange...

Reply:
@dantegd can you share your libraries installed in your conda environment?

Reply:
@xmnlab  i don't think that's necessary, you can repro by creating pygdf dev env from the README and then installing numpy=1.12

Reply:
yeah, just updating numpy to 1.14 solves it. That version (along numba 0.34) are part of the conda environment in libgdf, that's how they ended in the environment.

Reply:
@dantegd i just pushed a fix